pkg/services/apiserver/builder/admission.go

```diff
@@ -16,9 +17,24 @@ var _ admission.MutationInterface = (*builderAdmission)(nil)
 var _ admission.ValidationInterface = (*builderAdmission)(nil)
 
 func (b *builderAdmission) Admit(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	// on create, if the object does not have a name or a generate name specified, set one
+	// the name is the grafana uid, so we should not fail if it is not set
+	if a.GetOperation() == admission.Create {
+		obj := a.GetObject()
+		meta, err := utils.MetaAccessor(obj)
+		if err != nil {
+			return err
+		}
+		if meta.GetName() == "" && meta.GetGenerateName() == "" {
+			// a unique uid will be added to the end of it
+			meta.SetGenerateName("g")
```

Comment:
is it ok that the behavior here won't match CRD based APIs? it might be worth moving this out of the builder and into a standalone admission plugin ([example](https://github.com/kubernetes/kubernetes/blob/3a812ec4648967199e3079475956bba4b6bdf475/plugin/pkg/admission/admit/admission.go#L27C1-L46C2)) under `pkg/apiserver` so that cloud app platform's API server can use it if should be the default behavior for all APIs